src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -792,78 +793,87 @@ protected void calculateLengthFromGeometry () {
+            e1.length_mm = 1;
+        }
+        if (e2.length_mm < 0) {
+            LOG.error("Edge 2 ({}) split at vertex at {},{}  has length {} mm. Setting to 1 mm.", e2.wayId, v.getLat(), v.getLon(), e2.length_mm);
+            e2.length_mm = 1;
+        }
 
-            // TODO: better handle this temporary fix to handle bad edge distance calculation
-            if (e1.length_mm < 0) {
-                LOG.error("Edge 1 ({}) split at vertex at {},{} has length {} mm. Setting to 1 mm.", e1.wayId, v.getLat(), v.getLon(), e1.length_mm);
-                e1.length_mm = 1;
-            }
-            if (e2.length_mm < 0) {
-                LOG.error("Edge 2 ({}) split at vertex at {},{}  has length {} mm. Setting to 1 mm.", e2.wayId, v.getLat(), v.getLon(), e2.length_mm);
-                e2.length_mm = 1;
-            }
+        if (e1.length_mm < 0 || e2.length_mm < 0) {
+            e1.tov.removeIncoming(e1);
+            e1.fromv.removeOutgoing(e1);
+            e2.tov.removeIncoming(e2);
+            e2.fromv.removeOutgoing(e2);
+            throw new IllegalStateException("Split street is longer than original street!");
+        }
 
-            if (e1.length_mm < 0 || e2.length_mm < 0) {
-                e1.tov.removeIncoming(e1);
-                e1.fromv.removeOutgoing(e1);
-                e2.tov.removeIncoming(e2);
-                e2.fromv.removeOutgoing(e2);
-                throw new IllegalStateException("Split street is longer than original street!");
-            }
+        for (StreetEdge e : new StreetEdge[] { e1, e2 }) {
+            e.setBicycleSafetyFactor(getBicycleSafetyFactor());
+            e.setHasBogusName(hasBogusName());
+            e.setStairs(isStairs());
+            e.setWheelchairAccessible(isWheelchairAccessible());
+            e.setBack(isBack());
+        }
 
-            for (StreetEdge e : new StreetEdge[] { e1, e2 }) {
-                e.setBicycleSafetyFactor(getBicycleSafetyFactor());
-                e.setHasBogusName(hasBogusName());
-                e.setStairs(isStairs());
-                e.setWheelchairAccessible(isWheelchairAccessible());
-                e.setBack(isBack());
-            }
+        return new P2<>(e1, e2);
+    }
+
+    /** Split this street edge and return the resulting street edges. The original edge is kept. */
+    public P2<StreetEdge> splitNonDestructively(SplitterVertex v, DisposableEdgeCollection tempEdges) {
+        P2<LineString> geoms = GeometryUtils.splitGeometryAtPoint(getGeometry(), v.getCoordinate());
+
+        StreetEdge e1 = null;
+        StreetEdge e2 = null;
+
+        if (((TemporarySplitterVertex) v).isEndVertex()) {
+            e1 = new TemporaryPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
+            e1.setNoThruTraffic(this.isNoThruTraffic());
+            e1.setStreetClass(this.getStreetClass());
```

Comment:
This doesn't copy all the attributes that the destructive split does and therefore may not copy some critical information. In the IBI fork ([see here](https://github.com/ibi-group/OpenTripPlanner/blob/65dcf0a4142e31028cf9d1b2c15ad32dd1084252/src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java#L1479-L1498)), we added a method called `copyAttributres` that was used in both destructive and non-destructive splits to copy over all attributes not related to the length of the edge.